zigen-project/zen-oculus-display-system
# create XrInstance

## Context

Continuation of #3 .

In OpenXR we must create XrInstance first.
https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#instance
> An OpenXR instance is an object that allows an OpenXR application to communicate with an OpenXR runtime. The application accomplishes this communication by calling [xrCreateInstance](https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#xrCreateInstance) and receiving a handle to the resulting [XrInstance](https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#XrInstance) object.

## Summary

- [x] Define OpenXR context, which will include OpenXR instance, session, etc.
- [x] Create OpenXR instance and store it in the context 

## How to check behavior

1. Build and run with your Oculus Quest.
2. Confirm that no error log happens from "ZEN".
When some error happens, you will see
```log
E/ZEN: <some error log>
```
3. Checkout XrInstance info in the log
```log
I/ZEN: Instance RuntimeName=Oculus RuntimeVersion=43.178.0
```
